apps/frontend/src/hooks/use-app-update.ts

```
@@ -0,0 +1,59 @@
+import { useEffect, useRef, useCallback } from "react"
+import { toast } from "sonner"
+import { usePageActivity } from "./use-page-activity"
+import { useSocketReconnectCount } from "@/contexts/socket-context"
+
+const POLL_INTERVAL = 300_000 // 5 minutes
+const TOAST_ID = "app-update"
+
+export function useAppUpdate(): void {
+  if (__APP_VERSION__ === "dev") return
+
+  const toastShownRef = useRef(false)
```

Biome (2.4.4)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)